editor/selectors.js

```diff
@@ -226,9 +242,26 @@ export function isEditedPostBeingScheduled( state ) {
  * @return {String}       Raw post title
  */
 export function getEditedPostTitle( state ) {
-	return state.editor.edits.title === undefined
-		? get( state.currentPost, 'title.raw' )
-		: state.editor.edits.title;
+	const editedTitle = get( state.editor, 'edits.title' );
+	if ( editedTitle !== undefined ) {
+		return editedTitle;
+	}
+	return get( state.currentPost, 'title.raw' );
+}
+
+/**
+ * Gets the document title to be used.
+ *
+ * @param  {Object}  state Global application state
+ * @return {string}        Document title
+ */
+export function getDocumentTitle( state ) {
+	let title = getEditedPostTitle( state );
+
+	if ( ! title || '' === title.trim() ) {
```

Comment:
Maybe `getEditedPostTitle` should return an empty string as its default? In which case we could consolidate this as `if ( ! title.trim() ) {` . Not sure if there's a performance benefit to testing truthiness before a trim, but readability benefit in testing in single pass.